dist/src/main/java/io/camunda/commons/clustering/DynamicClusterServices.java

```diff
@@ -37,31 +39,31 @@ public DynamicClusterServices(final ActorScheduler scheduler, final AtomixCluste
   }
 
   @Bean
-  GatewayClusterConfigurationService gatewayClusterTopologyService() {
+  @Profile("!broker")
+  public GatewayClusterConfigurationService gatewayClusterTopologyService(
+      final BrokerTopologyManager brokerTopologyManager) {
     final var service =
         new GatewayClusterConfigurationService(
             clusterCommunicationService,
             clusterMembershipService,
             new ClusterConfigurationGossiperConfig(
                 false, Duration.ofSeconds(10), Duration.ofSeconds(1), 2));
     scheduler.submitActor(service).join();
+    service.addUpdateListener(brokerTopologyManager);
     return service;
   }
 
   @Bean
-  BrokerTopologyManager brokerTopologyManager(
-      @Autowired final GatewayClusterConfigurationService gatewayClusterTopologyService) {
+  public BrokerTopologyManager brokerTopologyManager() {
     final var brokerTopologyManager =
         new BrokerTopologyManagerImpl(clusterMembershipService::getMembers);
     scheduler.submitActor(brokerTopologyManager).join();
     clusterMembershipService.addListener(brokerTopologyManager);
-    gatewayClusterTopologyService.addUpdateListener(brokerTopologyManager);
```

Comment:
💭 So before it happened in all cases. But this was ok because it was part of the gateway package. 

Now we build the bean always, but not register the listener when we are broker. Does it make sense to build the bean in the first place when being a broker? 🤔

Reply:
> Does it make sense to build the bean in the first place when being a broker?

The bean was also created when being a broker: https://github.com/camunda/camunda/pull/19136/files/483d3fe34b5f110ca3120a08db604eaafb1a9aa1#diff-fcb76f33e9bc6cf0b2f74d3516a87cd86b658a19dc51172773832c294082f94cL38-L44

I kept it that way, and only if the app does not include the broker the listener gets registered (in the bean above).